presto-main/src/test/java/com/facebook/presto/block/TestBlockBuilder.java

```diff
@@ -80,6 +83,28 @@ public void testNewBlockBuilderLike()
             assertEquals(newPageBuilder.getBlockBuilder(i).getPositionCount(), 0);
             assertTrue(newPageBuilder.getBlockBuilder(i).getRetainedSizeInBytes() < pageBuilder.getBlockBuilder(i).getRetainedSizeInBytes());
         }
+
+        BlockBuilder newBigintBlockBuilder = bigintBlockBuilder.newBlockBuilderLike(null, 200);
+        assertEquals(newBigintBlockBuilder.getPositionCount(), 0);
+        assertEquals(newBigintBlockBuilder.getRetainedSizeInBytes(), 80);
+        newBigintBlockBuilder.writeLong(0);
+        assertEquals(newBigintBlockBuilder.getPositionCount(), 1);
+        assertEquals(newBigintBlockBuilder.getRetainedSizeInBytes(), 1880);
```

Comment:
curious, why does 1 long take 1800 bytes?

Reply:
@bhhari It's because the new `newBlockBuilderLike(null, 200)` reserved space for 200 elements. I added comments for these lines:
```
        // Reserved 200 longs and booleans for nulls array
        assertEquals(newBigintBlockBuilder.getRetainedSizeInBytes(), 1880);

        ...
        // Reserved 200 varchars of average length 5.9, and 201 ints for offsets and 200 booleans for nulls
        assertEquals(newVarcharBlockBuilder.getRetainedSizeInBytes(), 2360);

        ...
        // Reserved 200 varchars of average length 5.9, and 201 ints for offsets and 200 booleans for nulls
        assertEquals(newArrayBlockBuilder.getRetainedSizeInBytes(), 5848);
```